src/evict.c

```diff
@@ -520,7 +521,7 @@ int performEvictions(void) {
     monotime evictionTimer;
     elapsedStart(&evictionTimer);
 
-    while (mem_freed < mem_tofree) {
+    while (mem_freed < (ssize_t)mem_tofree) {
```

Comment:
do we want to consider a case where on 32bit system, redis is eating more than 2GB of ram, and decided it needs to free most of it (i.e mem_tofree > 2gb)?
in which case, maybe mem_freed needs to be `long long`? or not count negative deltas?

Reply:
hold on.. 
if there's a small sporadic negative delta, that's not that bad.. pretty soon other positive deltas will overcome it and mem_freed will become positive again, right?
so it's only a problem if it consistently gets negative deltas, right?

but another look at the code it actually seems that the initial report is wrong.
in case of negative delta it'll just exit immediately (not keep running the loop for a lot more than it should). making my claim above (about positive deltas eventually overcoming the rare negative delta) wrong.

is that right? if it is, then yes. let's make it `long long` and update the comment.

Reply:
it seems you're right... we're not talking about over-eviction we're talking about reaching OOM
i'll change to `long long` and update the comment